### PR TITLE
TS fixes: add SubIcon and Element types + remove default props

### PR DESCRIPTION
**Pull request by hadasfa**
#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/types/ElementContent.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { ReactElement } from "react";
+
+export type ElementContent = string | ReactElement;
```
Comment (niksa-monday) [inline src/types/ElementContent.ts:3]: Probably PR should have name:
`feat: add SubIcon and Element types` :)

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -9,16 +9,16 @@ import VibeComponentProps from "../../types/VibeComponentProps";
 import CloseIcon from "../Icon/Icons/components/CloseSmall";
 import SearchIcon from "../Icon/Icons/components/Search";
 import { NOOP } from "../../utils/function-utils";
-import { IconType } from "../Icon/IconConstants";
+import { Icon } from "src/types/SubIcon";
```
Comment (niksa-monday) [inline src/components/Search/Search.tsx:12]: [Optional]
We have three different 'conventions' for import type:
```
import { Icon } from "src/types/SubIcon";
import { SubIcon } from "../../types/SubIcon";
import { SubIcon } from "src/types/SubIcon";
```
Probably we can choose one...

File: src/components/AttentionBox/AttentionBox.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import Icon, { IconSubComponentProps } from "../Icon/Icon";
+import Icon from "../Icon/Icon";
 import IconButton from "../IconButton/IconButton";
 import CloseSmall from "../Icon/Icons/components/CloseSmall";
 import AlertIcon from "../Icon/Icons/components/Alert";
 import { IconType } from "../Icon/IconConstants";
 import { backwardCompatibilityForProperties } from "../../helpers/backwardCompatibilityForProperties";
 import { AttentionBoxType } from "./AttentionBoxConstants";
-import VibeComponentProps from "src/types/VibeComponentProps";
+import { SubIcon, VibeComponentProps } from "src/types";
```
Comment (sahariko) [inline src/components/AttentionBox/AttentionBox.tsx:10]: This kills internal packages that consume Vibe, import relatively, same for all other instances.

**Review by sahariko — CHANGES_REQUESTED** [review]

File: src/components/Avatar/Avatar.tsx [hunk 4]
```diff
@@ -48,33 +47,33 @@ export interface AvatarProps extends VibeComponentProps {
...
   src,
   icon,
   text,
   tooltipProps,
   ariaLabel,
-  withoutTooltip,
+  withoutTooltip = false,
   role,
-  backgroundColor,
+  backgroundColor = elementColorsNames.CHILI_BLUE,
```
Comment (sahariko) [inline src/components/Avatar/Avatar.tsx:61]: Why is this not capitalized?

File: src/components/Avatar/Avatar.tsx [hunk 5]
```diff
@@ -48,33 +47,33 @@ export interface AvatarProps extends VibeComponentProps {
 
 const Avatar: React.FC<AvatarProps> = ({
   id,
-  type,
+  type = AvatarTypes.TEXT,
   className,
-  textClassName,
-  size,
+  textClassName = "",
+  size = AvatarSizes.LARGE,
```
Comment (sahariko) [inline src/components/Avatar/Avatar.tsx:53]: It's customary to have enum names as plural, not singular, same for `AvatarTypes`.

File: src/components/AttentionBox/AttentionBox.tsx (hunk 3 above)
Comment (niksa-monday) [reply src/components/AttentionBox/AttentionBox.tsx:10]: I agree, but what interesting is how it's works before with:
```
import VibeComponentProps from "src/types/VibeComponentProps";
```

File: src/components/Avatar/Avatar.tsx (hunk 4 above)
Comment (hadasfa) [reply src/components/Avatar/Avatar.tsx:61]: old code, was like this before, not under this pr scope

File: src/components/Icon/Icon.tsx [hunk 6]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
 interface IconProps extends VibeComponentProps {
   // eslint-disable-next-line no-unused-vars
   onClick?: (event: React.MouseEvent) => void;
-  icon: string | React.FunctionComponent<IconSubComponentProps> | null;
+  /**
+   * we support three types of icons - SVG, FONT and SRC (classname) so this prop is either the name of the icon or the component
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:28]:
```suggestion
   * We support three types of icons - SVG, FONT and SRC (classname) so this prop is either the name of the icon or the component
```

File: src/components/Icon/Icon.tsx [hunk 7]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
+  icon: SubIcon;
+  /**
+   * Is icon is a button
+   */
   clickable?: boolean;
+  /**
+   * icon aria label support
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:36]:
```suggestion
   * Icon [aria label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) support
```

File: src/components/Icon/Icon.tsx [hunk 8]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
   iconLabel?: string;
+  /**
+   *  the type of the component - svg, font or custom svg (using react-inlinesvg)
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:40]:
```suggestion
   *  The type of the component - svg, font or custom svg (using [`react-inlinesvg`](https://github.com/gilbarbara/react-inlinesvg#readme))
```

File: src/components/Icon/Icon.tsx [hunk 9]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
   iconType?: IconType;
+  /**
+   * size for font icon
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:44]:
```suggestion
   * Size for font icon
```

File: src/components/Icon/Icon.tsx [hunk 10]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
   iconSize?: number | string;
+  /**
+   * remove focus style
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:48]:
```suggestion
   * Remove focus style
```

File: src/components/Icon/Icon.tsx [hunk 11]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
   ignoreFocusStyle?: boolean;
   tabindex?: number | string;
+  /**
+   * Hide icon asset from screen reader. No need to set value for this prop when clickable = false
+   */
   ariaHidden?: boolean;
   style?: React.CSSProperties;
+  /**
+   * when using svg from src (Icon.type.SRC) this boolean will transform the "fill" property to "currentColor"
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:58]:
```suggestion
   * When using svg from src (`Icon.type.SRC`) this boolean will transform the "fill" property to "currentColor"
```

File: src/components/Icon/Icon.tsx [hunk 12]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   * size for font icon
+   */
   iconSize?: number | string;
+  /**
+   * remove focus style
+   */
   ignoreFocusStyle?: boolean;
   tabindex?: number | string;
+  /**
+   * Hide icon asset from screen reader. No need to set value for this prop when clickable = false
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:53]:
```suggestion
   * Hide icon asset from screen reader. No need to set value for this prop when `clickable` is false
```

File: src/components/Icon/Icon.tsx [hunk 13]
```diff
@@ -26,16 +24,43 @@ export interface IconSubComponentProps {
...
+   */
   ariaHidden?: boolean;
   style?: React.CSSProperties;
+  /**
+   * when using svg from src (Icon.type.SRC) this boolean will transform the "fill" property to "currentColor"
+   */
   useCurrentColor?: boolean;
+  /**
+   * If you want to override to coloring of currentColor
```
Comment (sahariko) [inline src/components/Icon/Icon.tsx:62]:
```suggestion
   * Override the default color with a custom one
```

**Review by sahariko — APPROVED** [review]

**Pull request merged**